clusterloader2/api/validation.go

```diff
@@ -18,26 +18,230 @@ package api
 
 import (
 	"fmt"
+	"os"
 
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/perf-tests/clusterloader2/pkg/errors"
 )
 
+// ConfigValidator contains metadata for config validation.
+type ConfigValidator struct {
+	configDir string
+	config    *Config
+}
+
+// NewConfigValidator creates a new ConfigValidator object
+func NewConfigValidator(configDir string, config *Config) *ConfigValidator {
+	return &ConfigValidator{
+		configDir: configDir,
+		config:    config,
+	}
+}
+
 // Validate checks and verifies the configuration parameters.
-func (conf *Config) Validate() *errors.ErrorList {
-	// TODO(#1636): Validate other aspects of the api
-	return conf.Namespace.Validate()
+func (v *ConfigValidator) Validate() *errors.ErrorList {
+	allErrs := field.ErrorList{}
+	c := v.config
+
+	// TODO(#1696): Clean up after removing automanagedNamespaces
+	if c.AutomanagedNamespaces < 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("automanagedNamespaces"), c.AutomanagedNamespaces, "must be non-negative"))
+	}
+
+	allErrs = append(allErrs, v.validateNamespace(&c.Namespace, field.NewPath("namespace"))...)
+
+	for i := range c.TuningSets {
+		allErrs = append(allErrs, v.validateTuningSet(c.TuningSets[i], field.NewPath("tuningSets").Index(i))...)
+	}
+
+	for i := range c.Steps {
+		allErrs = append(allErrs, v.validateStep(c.Steps[i], field.NewPath("steps").Index(i))...)
+	}
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+
+	return v.toErrorList(allErrs)
 }
 
-// Validate checks and verifies the namespace parameters.
-func (ns *NamespaceConfig) Validate() *errors.ErrorList {
+func (v *ConfigValidator) toErrorList(fldErrors field.ErrorList) *errors.ErrorList {
 	errList := errors.NewErrorList()
+	for _, fldError := range fldErrors {
+		errList.Append(fmt.Errorf(fldError.Error()))
```

Comment:
You can just do:

errList.Append(fldError)

right?

Reply:
Ah that's right. Updated